src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -3,25 +3,26 @@
 
 using System.Buffers.Binary;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Reflection.Metadata;
-using System.Runtime.InteropServices;
 
 namespace System.Reflection.Emit
 {
     internal sealed class MethodBuilderImpl : MethodBuilder
     {
-        private readonly Type _returnType;
-        private readonly Type[]? _parameterTypes;
+        private Type _returnType;
+        private Type[]? _parameterTypes;
         private readonly ModuleBuilderImpl _module;
         private readonly string _name;
         private readonly CallingConventions _callingConventions;
         private readonly TypeBuilderImpl _declaringType;
         private MethodAttributes _attributes;
         private MethodImplAttributes _methodImplFlags;
 
+        private GenericTypeParameterBuilderImpl[]? _typeParameters;
+        private bool _isGenericMethod;
```

Comment:
This bool flag looks redundant. Can we use `_typeParameters != null` to detect whether the method is generic?